### PR TITLE
remove REDISMODULE_POSTPONED_ARRAY_LEN from serializeResult

### PR DESCRIPTION
**Pull request by ashtul**
An optimization that will determine the number of fields to return instead of using the `REDISMODULE_POSTPONED_ARRAY_LEN` flag.

### CODE REVIEW

**Comment by filipecosta90** [comment]
benchmark command (anonymizing data) :
```
redis-benchmark -e "FT.AGGREGATE" "idx:AAAA" "@field1:XX @field2:LL"
```

## v1.6.15 c30ee5b08c8fe59216f057cec239389f048d33e5
```
Summary:
  throughput summary: 7095.22 requests per second
  latency summary (msec):
          avg       min       p50       p95       p99       max
        6.954     0.360     6.983     7.207     7.343     8.863
```

## master 521044b1864372fe87752f8e30bfd3f186f2db57
```
Summary:
  throughput summary: 3215.54 requests per second
  latency summary (msec):
          avg       min       p50       p95       p99       max
       15.394     0.200    15.239    15.615    23.071    24.015
``` 

## remove-REDISMODULE_POSTPONED_ARRAY_LEN-from-serializeResult,

```
Summary:
  throughput summary: 6927.13 requests per second
  latency summary (msec):
          avg       min       p50       p95       p99       max
        7.125     0.240     7.159     7.367     7.447     8.239
```

File: src/rlookup.c [hunk 1]
```diff
@@ -106,10 +106,12 @@ RLookupKey *RLookup_GetKey(RLookup *lookup, const char *name, int flags) {
   return RLookup_GetKeyEx(lookup, name, strlen(name), flags);
 }
 
-size_t RLookup_GetLength(const RLookup *lookup, const RLookupRow *r, int requiredFlags,
-                         int excludeFlags) {
+size_t RLookup_GetLength(const RLookup *lookup, const RLookupRow *r, int *skipFieldIndex,
+                         int requiredFlags, int excludeFlags, SchemaRule *rule) {
+  int i = -1; // we do i++ right away and we don't want to skip i == 0
   size_t nfields = 0;
   for (const RLookupKey *kk = lookup->head; kk; kk = kk->next) {
```
Comment (MeirShpilraien) [inline src/rlookup.c:113]: You can do:
`for (const RLookupKey *kk = lookup->head, i = 0; kk; kk = kk->next, ++i)`
IIRC

**Review by MeirShpilraien — APPROVED** [review]
One small comment (not a must)

File: src/rlookup.c (hunk 1 above)
Comment (ashtul) [reply src/rlookup.c]: Nice touch!

**Review by MeirShpilraien — APPROVED** [review]

**Pull request merged**